### PR TITLE
Add Python Poetry documentation

### PR DESCRIPTION
**Pull request by julienp**
### Proposed changes

We now have native `poetry` support to manage virtual enviroments and
dependencies.

### Related issues (optional)

Ref: https://github.com/pulumi/pulumi/issues/15937

### CODE REVIEW

File: content/docs/languages-sdks/python/_index.md [hunk 1]
```diff
@@ -149,9 +136,42 @@ $ venv/bin/pip install -r requirements.txt
...
+#### Poetry
+
+To add a new dependency when using `poetry`, run the `poetry add` command in your project directory:
+
+{{< chooser os "macos,windows,linux" >}}
+
+{{% choosable os macos %}}
+
+```bash
+$ poetry add ${PACKAGE_NAME}
```
Comment (julienp) [inline content/docs/languages-sdks/python/_index.md:148]: I don't have a concrete package name to use here, are we ok with using syntax like `${PACKAGE_NAME}` to indicate that this is a place holder?

Comment (cnunciato) [reply content/docs/languages-sdks/python/_index.md:148]: This is okay, but for command-line args, we typically use this notation, with angle-brackets and hyphens:

```
poetry add <package-name>
```

So I'd suggest sticking with that if possible.

Comment (cnunciato) [reply content/docs/languages-sdks/python/_index.md:148]: I think this is fine, yeah, although we don't have a particular standard around this yet (and probably should). I'll poll the team and see what they think, but IMO this is good for now.

File: layouts/shortcodes/install-python.html [hunk 2]
```diff
@@ -11,12 +11,13 @@
     </div>
     <div class="content">
         <p>
-            <code>pip</code> is required to install dependencies. If you installed Python from source, with an installer from
+            <code>pip</code> or <code>poetry</code> are required to install dependencies. If you installed Python from source, with an installer from
```
Comment (cnunciato) [inline layouts/shortcodes/install-python.html:14]:
```suggestion
            Either <code>pip</code> or <code>poetry</code> is required to install dependencies. If you installed Python from source, with an installer from
```

**Review by cnunciato — APPROVED** [review]
Looks great! Nothing need block, just a few minor suggestions for improvement.

**Pull request merged**